utils/formatNumber.tsx

```diff
@@ -0,0 +1,15 @@
+export function formatLargeNumber(input: number | string, decimalPlaces: number = 2): string {
+    const num = typeof input === 'string' ? parseFloat(input) : input;
+  
+    if (isNaN(num)) return "Invalid number";
```

Comment:
"invalid number" will be bigger than the space we might have

I would return `"-"` or similar